shell: fix double new line print for RTT backend

jakub-uC opened the pull request:
In case terminal sends \r\n once user press Enter
shell will go to new line twice and it will print prompt
twice. This patch fixes it.

Signed-off-by: Jakub Rzeszutko <jakub.rzeszutko@nordicsemi.no>

nordic-krch commented on subsys/shell/shell.c:
```diff
@@ -45,6 +45,21 @@ static int shell_execute(const struct shell *shell);
 
+static inline bool ignore_nl(const struct shell *shell, uint8_t new_line)
+{
+	if ((shell->ctx->internal.flags.last_nl != '\r') &&
```
don't you need to assign every byte to last_nl to detect consecutive \r and \n?

jarz-nordic replied:
It is enough to save last state once, when commented condition is met.
Last new line character is reset when any character other than NL is received.

Commit:
shell: fix double new line print for RTT backend

In case terminal sends `\r\n` on the Enter button
shell will go to the new line twice and it will print
prompt twice. This patch fixes it.

Signed-off-by: Jakub Rzeszutko <jakub.rzeszutko@nordicsemi.no>

jarz-nordic commented on subsys/shell/shell.c:
```diff
@@ -45,6 +45,21 @@ static int shell_execute(const struct shell *shell);
 
+static inline bool ignore_nl(const struct shell *shell, uint8_t new_line)
+{
+	if ((shell->ctx->internal.flags.last_nl != '\r') &&
```
reworked

nordic-krch commented on subsys/shell/shell.c:
```diff
@@ -45,6 +45,22 @@ static int shell_execute(const struct shell *shell);
 extern const struct shell_cmd_entry __shell_root_cmds_start[0];
 extern const struct shell_cmd_entry __shell_root_cmds_end[0];
 
+static bool process_nl(const struct shell *shell, uint8_t data)
```
could you add doxygen comment with short description and return meaning.
